manifests/resource/streamhost.pp

```diff
@@ -47,23 +47,23 @@
 #    ensure   => present,
 #  }
 define nginx::resource::streamhost (
-  Enum['absent', 'present'] $ensure       = 'present',
-  Variant[Array, String] $listen_ip       = '*',
-  Integer $listen_port                    = 80,
-  Optional[String] $listen_options        = undef,
-  Boolean $ipv6_enable                    = false,
-  Variant[Array, String] $ipv6_listen_ip  = '::',
-  Integer $ipv6_listen_port               = 80,
-  String $ipv6_listen_options             = 'default ipv6only=on',
-  $proxy                                  = undef,
-  String $proxy_read_timeout              = $::nginx::proxy_read_timeout,
-  $proxy_connect_timeout                  = $::nginx::proxy_connect_timeout,
-  Array $resolver                         = [],
-  $raw_prepend                            = undef,
-  $raw_append                             = undef,
-  String $owner                           = $::nginx::global_owner,
-  String $group                           = $::nginx::global_group,
-  String $mode                            = $::nginx::global_mode,
+  Enum['absent', 'present'] $ensure                           = 'present',
+  Variant[Enum['*'], Array[String[1]], String[1]] $listen_ip  = '*',
+  Stdlib::Port $listen_port                                   = 80,
+  Optional[String[1]] $listen_options                         = undef,
+  Boolean $ipv6_enable                                        = false,
+  Variant[Array[Stdlib::Ipv6], Stdlib::Ipv6] $ipv6_listen_ip  = '::',
+  Stdlib::Port $ipv6_listen_port                              = 80,
+  String[1] $ipv6_listen_options                              = 'default ipv6only=on',
+  Optional[String[1]] $proxy                                  = undef,
+  Nginx::Duration $proxy_read_timeout                         = $::nginx::proxy_read_timeout,
+  Nginx::Duration $proxy_connect_timeout                      = $::nginx::proxy_connect_timeout,
+  Array[String[1]] $resolver                                  = [],
```

Comment:
now resolver is a required parameter, it needs to contain a string, this wasn't the case before. Should we change it back?